### PR TITLE
Support sharing via native iOS app

### PR DESCRIPTION
**Pull request by RalfNieuwenhuizen**

### CODE REVIEW

**Comment by Eyesonly88** [comment]
Hi @RalfNieuwenhuizen what's the difference between this PR and the social sharing dialog documented at 👉 https://developers.facebook.com/docs/react-native/sharing ?

**Comment by RalfNieuwenhuizen** [comment]
This change allows the selection of the native `FBSDKShareDialogModeNative` option, allowing more configuration of the sharing dialog you mention.

**Comment by Eyesonly88** [comment]
Okay cool, thanks @RalfNieuwenhuizen. Hey @lepouya or @muksit, can someone approve/merge this PR please? seems to be useful and small and has passed all the bot checks ✅ ...

File: ios/RCTFBSDK/share/RCTFBSDKShareDialog.m [hunk 1]
```diff
@@ -61,8 +62,9 @@ + (BOOL)requiresMainQueueSetup
 
 RCT_EXPORT_METHOD(canShow:(RCTFBSDKSharingContent)content resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
 {
-  _shareDialog.shareContent = content;
```
Comment (janicduplessis) [inline ios/RCTFBSDK/share/RCTFBSDKShareDialog.m]: Could you explain this change?

**Comment by janicduplessis** [comment]
Looks good, just one small clarification and I can land this. Could you also rebase on master to fix conflicts?

**Pull request merged**